dotnet/src/SemanticKernel.Abstractions/Functions/KernelParameterMetadata.cs

```diff
@@ -39,6 +57,24 @@ public KernelParameterMetadata(KernelParameterMetadata metadata)
         this.IsRequired = metadata.IsRequired;
         this._parameterType = metadata._parameterType;
         this._schema = metadata._schema;
+        this._jsonSerializerOptions = metadata._jsonSerializerOptions;
```

Comment:
Yes, if JSOs are supplied via metadata, it makes it compatible, but this will only be known at runtime, not at compile time. So, I thought it's better to show the warnings than not, allowing the callers to make decisions themselves based on their specific context. If JSOs were supplied in the metadata, they can suppress the warning.